src/App.tsx

```diff
@@ -1,14 +1,63 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+interface AppProps {
+  debounceDelay: number;
+  hideDelay: number;
+}
+
+export const App: React.FC<AppProps> = ({ debounceDelay, hideDelay }) => {
+  const [text, setInput] = useState('');
+  const [onFocus, setOnFocus]
+    = useState<HTMLInputElement | null | boolean>(null);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [debouncedInput, setDebouncedInput] = useState('');
+
+  useEffect(() => {
+    const debounceTimeout = setTimeout(() => {
+      setDebouncedInput(text);
+    }, debounceDelay);
+
+    return () => {
+      clearTimeout(debounceTimeout);
+    };
+  }, [text, debounceDelay]);
+
+  const handleSuggestionClick = (person: Person) => {
+    setInput(person.name);
+    setSelectedPerson(person);
+  };
+
+  const handleOnFocus = () => {
+    setOnFocus(true);
+  };
+
+  const handleOnBlur: React.FocusEventHandler<HTMLInputElement> = () => {
+    setTimeout(() => {
+      if (text === '') {
+        setOnFocus(false);
+      }
+    }, hideDelay);
+  };
+
+  function filteredPeople(people: Person[], input: string) {
+    return people.filter(person => person.name
+      .toLowerCase()
+      .includes(input.toLowerCase()));
+  }
+
```

Comment:
Your riun this function twice - it may cause performance issues
Use variable instead on function to get result .
Note: No need to filter in case query is empty
```suggestion
  const filteredPeople =  people.filter(({name}) => name
      .toLowerCase()
      .includes(input.toLowerCase()))

```